### DISCUSSION TITLE
Global Variables

### DISCUSSION THREAD

**Discussion by salaslg** [BIPES/BIPES#73] [Q&A] [answered]
Hi, I'm just getting started with using BIPES to program a Raspberry Pi Pico and I can't figure out how to declare global variables. It seems like the variables default to local types. Can you provide some instructions? Thanks.

**Comment by gastmaier** [comment]
Hi, 
it should be automatic since all declared variables are globals.
For example, in [this sketch (#cd8727)](https://bipes.net.br/ide/#cd8727) we have the `power` and `water_temp` variables inside the `Timer` block.

![Screenshot from 2021-10-31 10-43-48](https://user-images.githubusercontent.com/2892061/139586618-6e167d99-41cb-46a6-87e1-0214fdb9c51e.png)

And when we look at the code, we can see that BIPES included the `global` keyword in line 39

![Screenshot from 2021-10-31 10-43-32](https://user-images.githubusercontent.com/2892061/139586703-e8bb653f-cbae-41d2-bcff-368da930af15.png)

However, if  you found some block that you think it should include the globals keyword and it didn't, please provide an example and I will look into it :smile:

**Reply by salaslg** [reply]
Thanks for replying so quickly. I've attached the example below.

<img width="515" alt="Screen Shot 2021-11-01 at 7 08 09 AM" src="https://user-images.githubusercontent.com/60052594/139671716-7fad40d4-1f75-42c6-bd0c-ea3017ab2b06.png">

The interrupt block updates a variable, button_pressed, but it's not defined as global, so the rest of the code can't use it.

<img width="600" alt="Screen Shot 2021-11-01 at 7 08 09 AM 1" src="https://user-images.githubusercontent.com/60052594/139672539-08f7c915-180b-47f8-8141-084aafbab98b.png">

**Comment by gastmaier** [comment, marked as answer]
Oh ok, the block `gpio_interrupt` wasn't including the global variables, but I just fixed it with 247fe6d82d5b76043e40eae91b2ad253d0dc6be2.
Check it out: https://bipes.net.br/ide/#8gukhm

Ps. Please refresh the IDE cache with `Ctrl+F5` so it'll fetch the new code.

**Reply by salaslg** [reply]
Thanks! That did the trick!
